File: src/components/Quiz/QASection/Answer/Answer.js

import classes from './Answer.module.css';

const answer = ({answerData, collectAnswerData, setAnswered, answered, ...rest}) => {
    // const [answered, setAnswered] = useState(false);

    const answerClicked = () => {
        if(answerData.correct) {
            console.log('correct');
        } else {
            console.log('incorrect');
        }
        setAnswered(answerData.correct);
        collectAnswerData(answerData.correct);
    }
    return (
        <button disabled={answered} className={classes.Answer} onClick={() => answerClicked()}>
            {answerData.value}
        </button>
    )
};


export default answer;

File: quizMeServer/quizMeServer/Models/Answer.cs
namespace quizMeServer.Models
{
    public class Answer
    {
        public string Id { get; set; }
        public string Value { get; set; }
        public bool Correct { get; set; }

        public Answer(string id, string value, bool correct)
        {
            Id = id;
            Value = value;
            Correct = correct;
        }
    }
}
